src/OmnichannelChatSDK.ts

```diff
@@ -227,6 +231,44 @@ class OmnichannelChatSDK {
         loggerUtils.setRequestId(this.requestId, this.ocSdkLogger, this.acsClientLogger, this.acsAdapterLogger, this.callingSdkLogger, this.amsClientLogger, this.ic3ClientLogger);
     }
 
+    /**
+     * Executes an operation with mutual exclusion to prevent race conditions
+     * between startChat and endChat operations
+     */
+    private async executeWithLock<T>(operation: () => Promise<T>): Promise<T> {
```

Comment:
What about a scenario where there are server problems with the start chat API via chatSDK? i.e. - startChat hangs (gets stuck). Is there a timeout scenario for this on the backend side so we do not accidentally block the pendingOperations queue?

I think for now this is fine, because only startChat and endChat are executing with the lock, and one cannot end a chat before starting it, but the reverse should also be considered. What if endChat hangs, and then the next conversation cannot call start chat because the mutex is locked on the previous endChat operation?

One way to resolve such problems might be to give executeWithLock a timeout mechanism, that way eventually the pendingOperations queue will clear, regardless of hanging.